e2e/src/pages/navigate.po.ts

```diff
@@ -1,20 +1,32 @@
-import { by, element } from 'protractor';
-import { AppPage } from './app.po';
+import {browser, by, element, ElementFinder} from 'protractor';
+import {AppPage} from './app.po';
 
-export class NavigatePage extends AppPage{
+export class NavigatePage extends AppPage {
+  public pageNumber: ElementFinder;
+  public next: ElementFinder;
+  public previous: ElementFinder;
 
-  async goToNextPage() {
-    await this.waitForElement(by.id('next'));
-    await this.clickElement(by.id('next'));
-  }
-
-  async goToPreviousPage() {
-    await this.waitForElement(by.id('previous'));
-    await this.clickElement(by.id('previous'));
+  constructor() {
+    super();
+    this.next = element(by.id('next'));
+    this.previous = element(by.id('previous'));
+    this.pageNumber = element(by.id('pageNumber'));
   }
 
   async number() {
     await this.waitForElement(by.id('pageNumber'));
     return element(by.id('pageNumber')).getAttribute('value');
   }
+
+  async goToNextPage() {
+    await this.next.click().then(async function () {
+      await browser.driver.sleep(2000);
+    });
+  }
+
+  async goToPreviousPage() {
+    await this.previous.click().then(async function () {
+      await browser.driver.sleep(2000);
+    });
+  }
```

Comment:
I don't get why this is being changed?
also, I would avoid using the sleep method if possible, as the time to wait changes depending on the environment
there are other mechanisms like waitForElement that we're using elsewhere (well, in the previous code)

Reply:
removed ES5 promise syntax and used ES7 sysntax